arch/x86_64/mcount-insn.c

```diff
@@ -78,6 +78,90 @@ static int opnd_reg(int capstone_reg)
 	return -1;
 }
 
+/*
+ * Handle relative jumps and conditional branches.
+ *
+ * This function relocates jcc8 and jcc32 instructions by replacing them with a jcc8 
+ * that has a null offset. The offset will be patched later when the code is saved for out
+ * of line execution. The new jcc8 will jump (if condition is met) to a trampoline
+ * which itself jumps to the target of the orginal instruction.
+ *
+ * The relocation of jmp8 and jmp32 is achieved by replacing them with an absolute 
+ * indirect jump to the target.
+ *
+ */
+static int handle_rel_jmp(cs_insn *insn, uint8_t insns[], struct mcount_disasm_info *info)
+{
+	cs_x86 *x86 = &insn->detail->x86;
+	uint8_t relocated_insn[ARCH_TRAMPOLINE_SIZE] = { 0xff, 0x25,};
+	uint8_t opcode = insn->bytes[0];
+	uint64_t target;
+	struct cond_branch_info *cbi;
+
+#define JMP8_OPCODE 0xEB
+#define JMP32_OPCODE 0xE9
+#define JCC8_SIZE   2
+#define IND_JMP_SIZE 6
+#define OP   0
+#define OFS	  1
+	cs_x86_op *opnd = &x86->operands[0];
+
+	if (x86->op_count != 1 || opnd->type != X86_OP_IMM)
+		goto out;
+
+	if ((opcode == JMP8_OPCODE || opcode == JMP32_OPCODE)) {
+		if (strcmp(insn->mnemonic, "jmp") != 0)
+			goto out;
+
+		target = opnd->imm;
+
+		memcpy(insns, relocated_insn, IND_JMP_SIZE);
+		memcpy(insns + IND_JMP_SIZE, &target, sizeof(target));
+
+		info->modified = true;
+		/* If this jump is the last insn in the prologue, we can ignore
+		 * the one in patch_normal_func()
+		 */
```

Comment:
It's just a style issue, but it'd be better being consistent.  Please start from the second line for multi-line comments.